Change comment formula of enet_coordinate_descent_gram

edwingrappin opened the pull request:
I changed the formula in the comment of enet_coordinate_descent_gram function. 
Indeed, the function solves w^T Q w - 2 q^T w + alpha norm(w, 1) + beta norm(w, 2)^2 instead of w^T Q w - q^T w + alpha norm(w, 1) + beta norm(w, 2)^2 (there was a factor 2 missing in front of the linear term q^T w).

Commit:
Change comment formula of enet_coordinate_descent_gram 

I changed the formula in the comment of enet_coordinate_descent_gram function. 
Indeed, the function solves w^T Q w - 2 q^T w + alpha norm(w, 1) + beta norm(w, 2)^2 instead of w^T Q w - q^T w + alpha norm(w, 1) + beta norm(w, 2)^2 (there was a factor 2 missing in front of the linear term q^T w).

rvraghav93 commented:
Thanks LGTM

**cc:** @MechCoder @agramfort

agramfort commented on sklearn/linear_model/cd_fast.pyx:
```diff
@@ -498,7 +498,7 @@ def enet_coordinate_descent_gram(double[:] w, double alpha, double beta,
         We minimize
 
-        1 w^T Q w - q^T w + alpha norm(w, 1) + beta norm(w, 2)^2
+        1 w^T Q w - 2 q^T w + alpha norm(w, 1) + beta norm(w, 2)^2
```
I think it was correct before. You have the 1/2 in front of the other terms

TomDLT commented:
At least this PR shows that the way the 1/2 are written is not clear (!)

agramfort commented:
PR welcome to clarify :)

edwinENSAE commented:
Oh I just understand that : 
        1 w^T Q w - q^T w + alpha norm(w, 1) + beta norm(w, 2)^2
        -                                      ----
        2                                        2


means 
        1/2 w^T Q w - q^T w + alpha norm(w, 1) + beta/2 norm(w, 2)^2

I am not familiar with the first style of comment, but I believe it is the way the community is used to commenting it? I am not familiar with computer science habits (yet). Sorry.

rvraghav93 commented:
This could be rewritten like this I feel

```
         1                                       beta
        --- w^T Q w - q^T w + alpha norm(w, 1) + ---- norm(w, 2)^2
         2                                        2

```

rvraghav93 commented:
One could also use the unicode dash, but that would be taking this a bit overboard ;) (And also we don't usually encourage unicode in our codebase I think)

```
     1
    ―――
     2
```

edwinENSAE commented:
@rvraghav93 , to my point of view, your first proposition is good:
         1                                                               beta           
        --- w^T Q w - q^T w + alpha norm(w, 1) + ---- norm(w, 2)^2
         2                                                                 2

agramfort commented:
@rvraghav93 +1

amueller commented:
or we could use latex? ;) (but also the first is good. or just write 1/2 ... + beta/2 ? what's wrong with that?

agramfort commented:
just write 1/2 is good for me

On Fri, Jan 15, 2016 at 10:03 PM, Andreas Mueller <notifications@github.com>
wrote:

> or we could use latex? ;) (but also the first is good. or just write 1/2
> ... + beta/2 ? what's wrong with that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/6155#issuecomment-172091853>
> .
>

Commit:
Change coefficient in comments formula

I changed the formula as proposed by agramfort and added consistency in the four functions comments.

MechCoder commented:
Merged with master. Thanks :-)
